Sources/Optimizely/OptimizelyClient.swift

```diff
@@ -139,80 +149,74 @@ open class OptimizelyClient: NSObject {
     ///   - datafile: This datafile will be used when cached copy is not available (fresh start)
     ///             A cached copy from previous download is used if it's available.
     ///             The datafile will be updated from the server in the background thread.
+    ///   - doUpdateConfigOnNewDatafile: When a new datafile is fetched from the server in the background thread, the SDK will be updated with the new datafile immediately if this value is set to true. When it's set to false (default), the new datafile is cached and will be used when the SDK is started again.
     ///   - doFetchDatafileBackground: This is for debugging purposes when
     ///             you don't want to download the datafile.  In practice, you should allow the
     ///             background thread to update the cache copy (optional)
-    public func start(datafile: Data, doFetchDatafileBackground: Bool = true) throws {
+    public func start(datafile: Data,
+                      doUpdateConfigOnNewDatafile: Bool = false,
```

Comment:
These two don't really make sense here.  If you have doUpdateConfigOnNewDatafile but you have doFeatchDatafileBackground as false.  It is an invalid combination.  Maybe an enum might be better here with all valid combinations.  Also, the default combination is probably not ideal if you have polling enabled which is not addressed in this combo or PR.  So, you might want combos that include polling and what to do there.

Reply:
Yes it's not so clean as it should be. The reason I keep "doFetchDatafileBackground" is for API backward compatibility only, where we had this option for debugging support. We can let them override "doUpdateConfigOnNewDatafile" when they still want to disable fetch completely for debugging purpose.